Address safer CPP warnings in WebViewImpl.mm

csaavedra opened the pull request:
#### 110713922f458b43a98b6ba2feefd803e2333650
<pre>
Address safer CPP warnings in WebViewImpl.mm
<a href="https://bugs.webkit.org/show_bug.cgi?id=299872">https://bugs.webkit.org/show_bug.cgi?id=299872</a>

Reviewed by Chris Dumez.

Incidentally also fixed a few warnings in PageClientImplMac.mm, but
that file still has plenty of its own.

Renamed some methods to add a singleton suffix, others were made into
protected variants, added a few protected getters, etc.

* Source/WebKit/SaferCPPExpectations/RetainPtrCtorAdoptCheckerExpectations:
* Source/WebKit/Shared/mac/PasteboardTypes.h:
* Source/WebKit/Shared/mac/PasteboardTypes.mm:
(WebKit::PasteboardTypes::forEditingSingleton):
(WebKit::PasteboardTypes::forURLSingleton):
(WebKit::PasteboardTypes::forImagesSingleton):
(WebKit::PasteboardTypes::forImagesWithArchiveSingleton):
(WebKit::PasteboardTypes::forSelectionSingleton):
(WebKit::PasteboardTypes::forEditing): Deleted.
(WebKit::PasteboardTypes::forURL): Deleted.
(WebKit::PasteboardTypes::forImages): Deleted.
(WebKit::PasteboardTypes::forImagesWithArchive): Deleted.
(WebKit::PasteboardTypes::forSelection): Deleted.
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::isFullScreen):
(WebKit::PageClientImpl::enterFullScreen):
(WebKit::PageClientImpl::exitFullScreen):
(WebKit::PageClientImpl::beganEnterFullScreen):
(WebKit::PageClientImpl::beganExitFullScreen):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(-[WKAccessibilitySettingsObserver _settingsDidChange:]):
(-[WKWindowVisibilityObserver _windowDidOrderOnScreen:]):
(-[WKWindowVisibilityObserver _windowDidOrderOffScreen:]):
(-[WKWindowVisibilityObserver _windowDidBecomeKey:]):
(-[WKWindowVisibilityObserver _windowDidResignKey:]):
(-[WKWindowVisibilityObserver _windowDidMiniaturize:]):
(-[WKWindowVisibilityObserver _windowDidDeminiaturize:]):
(-[WKWindowVisibilityObserver _windowDidMove:]):
(-[WKWindowVisibilityObserver _windowDidResize:]):
(-[WKWindowVisibilityObserver _windowWillBeginSheet:]):
(-[WKWindowVisibilityObserver _windowDidChangeBackingProperties:]):
(-[WKWindowVisibilityObserver _windowDidChangeScreen:]):
(-[WKWindowVisibilityObserver _windowDidChangeOcclusionState:]):
(-[WKWindowVisibilityObserver _windowWillClose:]):
(-[WKWindowVisibilityObserver _screenDidChangeColorSpace:]):
(-[WKWindowVisibilityObserver observeValueForKeyPath:ofObject:change:context:]):
(-[WKWindowVisibilityObserver _activeSpaceDidChange:]):
(-[WKDOMPasteMenuDelegate _web_grantDOMPasteAccess]):
(-[WKTextTouchBarItemController textListViewController]):
(-[WKImageAnalysisOverlayViewDelegate initWithWebViewImpl:]):
(-[WKImageAnalysisOverlayViewDelegate dealloc]):
(-[WKImageAnalysisOverlayViewDelegate observeValueForKeyPath:ofObject:change:context:]):
(-[WKImageAnalysisOverlayViewDelegate firstResponderIsInsideImageOverlay]):
(-[WKImageAnalysisOverlayViewDelegate contentsRectForImageAnalysisOverlayView:]):
(WebKit::m_flagsChangedEventMonitorTrackingArea):
(WebKit::WebViewImpl::window):
(WebKit::WebViewImpl::protectedWindow):
(WebKit::WebViewImpl::handleProcessSwapOrExit):
(WebKit::WebViewImpl::setDrawsBackground):
(WebKit::WebViewImpl::setBackgroundColor):
(WebKit::WebViewImpl::becomeFirstResponder):
(WebKit::WebViewImpl::resignFirstResponder):
(WebKit::WebViewImpl::showWarningView):
(WebKit::WebViewImpl::isFocused const):
(WebKit::WebViewImpl::createPDFHUD):
(WebKit::WebViewImpl::renewGState):
(WebKit::WebViewImpl::setFrameSize):
(WebKit::WebViewImpl::setFrameAndScrollBy):
(WebKit::WebViewImpl::updateWindowAndViewFrames):
(WebKit::WebViewImpl::updateLayer):
(WebKit::WebViewImpl::updateViewExposedRect):
(WebKit::WebViewImpl::setIntrinsicContentSize):
(WebKit::WebViewImpl::intrinsicDeviceScaleFactor const):
(WebKit::WebViewImpl::windowDidBecomeKey):
(WebKit::WebViewImpl::windowDidResignKey):
(WebKit::WebViewImpl::windowDidChangeScreen):
(WebKit::WebViewImpl::mightBeginDragWhileInactive):
(WebKit::WebViewImpl::acceptsFirstMouse):
(WebKit::WebViewImpl::shouldDelayWindowOrderingForEvent):
(WebKit::WebViewImpl::windowResizeMouseLocationIsInVisibleScrollerThumb):
(WebKit::WebViewImpl::viewWillMoveToWindowImpl):
(WebKit::WebViewImpl::viewDidMoveToWindow):
(WebKit::WebViewImpl::viewDidChangeBackingProperties):
(WebKit::WebViewImpl::pageDidScroll):
(WebKit::WebViewImpl::scrollViewFrame):
(WebKit::WebViewImpl::updateTitlebarAdjacencyState):
(WebKit::WebViewImpl::hitTest):
(WebKit::WebViewImpl::colorSpace):
(WebKit::WebViewImpl::updateSecureInputState):
(WebKit::WebViewImpl::notifyInputContextAboutDiscardedComposition):
(WebKit::WebViewImpl::fullScreenWindowController):
(WebKit::WebViewImpl::protectedFullScreenWindowController):
(WebKit::WebViewImpl::executeSavedCommandBySelector):
(WebKit::WebViewImpl::registerEditCommand):
(WebKit::WebViewImpl::clearAllEditCommands):
(WebKit::WebViewImpl::validRequestorForSendAndReturnTypes):
(WebKit::WebViewImpl::selectionDidChange):
(WebKit::protectedMenuItem):
(WebKit::WebViewImpl::validateUserInterfaceItem):
(WebKit::WebViewImpl::setUserInterfaceItemState):
(WebKit::WebViewImpl::handleRequestedCandidates):
(WebKit::WebViewImpl::handleAcceptedCandidate):
(WebKit::WebViewImpl::preferencesDidChange):
(WebKit::WebViewImpl::dismissContentRelativeChildWindowsWithAnimation):
(WebKit::WebViewImpl::dismissContentRelativeChildWindowsWithAnimationFromViewOnly):
(WebKit::WebViewImpl::dismissContentRelativeChildWindowsFromViewOnly):
(WebKit::WebViewImpl::hasContentRelativeChildViews const):
(WebKit::WebViewImpl::suppressContentRelativeChildViews):
(WebKit::WebViewImpl::restoreContentRelativeChildViews):
(WebKit::WebViewImpl::quickLookWithEvent):
(WebKit::WebViewImpl::setAllowsLinkPreview):
(WebKit::WebViewImpl::immediateActionAnimationControllerForHitTestResult):
(WebKit::WebViewImpl::prepareForImmediateActionAnimation):
(WebKit::WebViewImpl::cancelImmediateActionAnimation):
(WebKit::WebViewImpl::completeImmediateActionAnimation):
(WebKit::WebViewImpl::didChangeContentSize):
(WebKit::WebViewImpl::videoControlsManagerDidChange):
(WebKit::WebViewImpl::setIgnoresNonWheelEvents):
(WebKit::WebViewImpl::accessibilityRegisterUIProcessTokens):
(WebKit::WebViewImpl::accessibilityAttributeValue):
(WebKit::WebViewImpl::updatePrimaryTrackingAreaOptions):
(WebKit::WebViewImpl::removeTrackingRect):
(WebKit::WebViewImpl::toolTipOwnerForSendingMouseEvents const):
(WebKit::WebViewImpl::sendToolTipMouseExited):
(WebKit::WebViewImpl::sendToolTipMouseEntered):
(WebKit::WebViewImpl::toolTipChanged):
(WebKit::WebViewImpl::updateThumbnailViewLayer):
(WebKit::WebViewImpl::setInspectorAttachmentView):
(WebKit::WebViewImpl::sendDragEndToPage):
(WebKit::WebViewImpl::draggingEntered):
(WebKit::WebViewImpl::draggingUpdated):
(WebKit::WebViewImpl::draggingExited):
(WebKit::handleLegacyFilesPromisePasteboard):
(WebKit::WebViewImpl::performDragOperation):
(WebKit::WebViewImpl::hitTestForDragTypes):
(WebKit::WebViewImpl::registerDraggedTypes):
(WebKit::WebViewImpl::didPerformDragOperation):
(WebKit::WebViewImpl::startWindowDrag):
(WebKit::WebViewImpl::startDrag):
(WebKit::WebViewImpl::setPromisedDataForImage):
(WebKit::WebViewImpl::provideDataForPasteboard):
(WebKit::pasteboardForAccessCategory):
(WebKit::WebViewImpl::requestDOMPasteAccess):
(WebKit::WebViewImpl::takeViewSnapshot):
(WebKit::WebViewImpl::setMagnification):
(WebKit::WebViewImpl::swipeWithEvent):
(WebKit::WebViewImpl::magnifyWithEvent):
(WebKit::WebViewImpl::smartMagnifyWithEvent):
(WebKit::WebViewImpl::gestureEventWasNotHandledByWebCore):
(WebKit::WebViewImpl::gestureEventWasNotHandledByWebCoreFromViewOnly):
(WebKit::WebViewImpl::collectKeyboardLayoutCommandsForEvent):
(WebKit::WebViewImpl::interpretKeyEvent):
(WebKit::WebViewImpl::doCommandBySelector):
(WebKit::WebViewImpl::firstRectForCharacterRange):
(WebKit::WebViewImpl::characterIndexForPoint):
(WebKit::WebViewImpl::inputContext):
(WebKit::WebViewImpl::performKeyEquivalent):
(WebKit::WebViewImpl::keyDown):
(WebKit::WebViewImpl::nativeMouseEventHandler):
(WebKit::WebViewImpl::removeFlagsChangedEventMonitor):
(WebKit::WebViewImpl::mouseMoved):
(WebKit::WebViewImpl::windowIsFrontWindowUnderMouse):
(WebKit::WebViewImpl::userInterfaceLayoutDirection):
(WebKit::WebViewImpl::effectiveAppearanceIsDark):
(WebKit::WebViewImpl::useFormSemanticContext const):
(WebKit::WebViewImpl::updateTouchBar):
(WebKit::WebViewImpl::dismissTextTouchBarPopoverItemWithIdentifier):
(WebKit::WebViewImpl::updateTextTouchBar):
(WebKit::WebViewImpl::updateMediaTouchBar):
(WebKit::WebViewImpl::shouldRequestCandidates const):
(WebKit::WebViewImpl::updateCursorAccessoryPlacement):
(WebKit::WebViewImpl::ensureImageAnalyzer):
(WebKit::WebViewImpl::installImageAnalysisOverlayView):

Canonical link: <a href="https://commits.webkit.org/301130@main">https://commits.webkit.org/301130@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e36d5c5aa83708935e5f8ba937fbf30fcd3599ae

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125023 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44692 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35429 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131872 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76900 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45384 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53258 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95168 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63142 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127977 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36221 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111810 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75711 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35140 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75354 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105984 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30193 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134548 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51845 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39644 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103639 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52270 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108022 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103413 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26331 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48750 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27040 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48891 "Hash e36d5c5a for PR 51562 does not build (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51730 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57523 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51104 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54461 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52795 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

csaavedra commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -349,20 +350,20 @@ - (void)startObserving:(NSWindow *)window
+    [defaultNotificationCenter addObserver:self selector:@selector(_windowDidDeminiaturize:) name:RetainPtr { NSWindowDidDeminiaturizeNotification }.get() object:window];
+    [defaultNotificationCenter addObserver:self selector:@selector(_windowDidMove:) name:RetainPtr { NSWindowDidMoveNotification }.get() object:window];
+    [defaultNotificationCenter addObserver:self selector:@selector(_windowDidResize:) name:RetainPtr { NSWindowDidResizeNotification }.get() object:window];
+    [defaultNotificationCenter addObserver:self selector:@selector(_windowWillBeginSheet:) name:RetainPtr { NSWindowWillBeginSheetNotification }.get() object:window];
```
These changes (and other of the sort) shouldn't be needed after https://github.com/llvm/llvm-project/pull/161146 lands, so I am not sure if we want them at all. I saw in a different MR that changes like this landed in the tree, so I am leaving them for now.

csaavedra commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -1394,7 +1397,7 @@ static NSTrackingAreaOptions flagsChangedEventMonitorTrackingAreaOptions()
     [m_layoutStrategy invalidate];
 
-    [m_immediateActionController willDestroyView:m_view.getAutoreleased()];
+    [m_immediateActionController willDestroyView: RetainPtr { view() }.get()];
```
This shouldn't be needed after https://github.com/llvm/llvm-project/pull/161236 lands.

csaavedra commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -1549,10 +1557,10 @@ static NSTrackingAreaOptions flagsChangedEventMonitorTrackingAreaOptions()
     // Predict the case where we are losing first responder status only to
     // gain it back again. We want resignFirstResponder to do nothing in that case.
-    RetainPtr<id> nextResponder = [[m_view window] _newFirstResponderAfterResigning];
+    RetainPtr<id> nextResponder = adoptNS([window() _newFirstResponderAfterResigning]);
```
Not entirely sure about this change, I couldn't find the implementation of `_newFirstResponderAfterResigning`, but the checker complains that a reference is leaked (maybe just due to the function naming). So please double check.

csaavedra replied:
This was not correct, it caused an exception. I suppressed it instead in a new a version.

csaavedra reviewed (commented):
I left some review comments for changes I am not entirely sure about, please check those.

cdumez commented on Source/WebKit/UIProcess/mac/WebViewImpl.h:
```diff
@@ -229,10 +229,12 @@ class WebViewImpl final : public CanMakeWeakPtr<WebViewImpl>, public CanMakeChec
     WebPageProxy& page() { return m_page.get(); }
 
     WKWebView *view() const { return m_view.getAutoreleased(); }
+    RetainPtr<WKWebView> protectedView() const { return view(); };
```
This should use `return m_view.get();`. Calling `view()` isn't great here because it will unnecessarily autorelease.

cdumez commented on Source/WebKit/UIProcess/mac/WebViewImpl.h:
```diff
@@ -907,7 +910,7 @@ class WebViewImpl final : public CanMakeWeakPtr<WebViewImpl>, public CanMakeChec
 
 #if ENABLE(IMAGE_ANALYSIS)
-    CocoaImageAnalyzer *ensureImageAnalyzer();
+    RetainPtr<CocoaImageAnalyzer> ensureProtectedImageAnalyzer();
```
This goes against our usual style. This should keep returning a raw pointer for call sites that do not need a smart pointer. And you can add a new `ensureProtectedImageAnalyzer()` for call sites that do need a smart pointer.

cdumez replied:
We should also mark `m_imageAnalyzer` as `const` and use `lazyInitialize()` to initialize it. It would make at least make warnings go away for code using `m_imageAnalyzer` directly.

csaavedra replied:
It seems that it's never used in a place where a smart pointer wouldn't be needed. But I am going to leave the original raw getter and remove the protected one for now, I think that after marking the member as `const` this warning will go away without needing a protected getter.

cdumez commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -1187,13 +1188,15 @@ - (instancetype)initWithWebViewImpl:(WebKit::WebViewImpl&)impl
 {
-    [_overlayView removeObserver:self forKeyPath:@"hasActiveTextSelection"];
+    RetainPtr overlayView { _overlayView };
+    [overlayView.get() removeObserver:self forKeyPath:@"hasActiveTextSelection"];
```
`.get()` is not needed.

cdumez commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -1786,11 +1795,11 @@ static NSTrackingAreaOptions flagsChangedEventMonitorTrackingAreaOptions()
 
         if (weakThis->m_needsViewFrameInWindowCoordinates)
-            viewFrameInWindowCoordinates = [weakThis->m_view convertRect:[weakThis->m_view frame] toView:nil];
+            viewFrameInWindowCoordinates = [weakThis->m_view.get() convertRect:[weakThis->m_view.get() frame] toView:nil];
```
We should store the view in a local variable to avoid ref'ing twice.

cdumez commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -6907,19 +6933,19 @@ static NSTextAlignment nsTextAlignmentFromTextAlignment(TextAlignment textAlignm
 {
     if (!m_imageAnalyzer) {
         m_imageAnalyzerQueue = WorkQueue::create("WebKit image analyzer queue"_s);
         m_imageAnalyzer = createImageAnalyzer();
```
Should likely use `lazyInitialize()` here and mark the data member as `const`, assuming it never gets reset.

csaavedra replied:
Looks like it's never reset.

cdumez commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -6219,10 +6243,10 @@ static _WKRectEdge toWKRectEdge(WebCore::RectEdges<bool> edges)
     NSInteger eventWindowNumber = [NSWindow windowNumberAtPoint:eventScreenPosition.origin belowWindowWithWindowNumber:0];
 
-    return [m_view window].windowNumber != eventWindowNumber;
+    return protectedWindow().get().windowNumber != eventWindowNumber;
```
We should store the window in a local variable to avoid reffing it twice.

cdumez commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -6102,7 +6126,7 @@ static BOOL shouldUseHighlightsForMarkedText(NSAttributedString *string)
 
     // When a view is first responder, it gets mouse moved events even when the mouse is outside its visible rect.
-    if (m_view.getAutoreleased() == [m_view window].firstResponder && !NSPointInRect([m_view convertPoint:[event locationInWindow] fromView:nil], [m_view visibleRect]))
+    if (m_view.getAutoreleased() == [m_view.get() window].firstResponder && !NSPointInRect([m_view.get() convertPoint:[event locationInWindow] fromView:nil], [m_view.get() visibleRect]))
```
We should store `m_view.get()` in a local to avoid so much ref'ing.

csaavedra replied:
This could be simplified further by using the local view instead of the autoreleased getter:

```diff
-    if (m_view.getAutoreleased() == [m_view.get() window].firstResponder && !NSPointInRect([m_view.get() convertPoint:[event locationInWindow] fromView:nil], [m_view.get() visibleRect]))
+    RetainPtr view = m_view.get();
+    if (view == [view window].firstResponder && !NSPointInRect([view convertPoint:[event locationInWindow] fromView:nil], [view visibleRect]))
```

cdumez commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -5450,10 +5473,10 @@ static bool eventKeyCodeIsZeroOrNumLockOrFn(NSEvent *event)
     LOG(TextInput, "characterIndexForPoint:(%f, %f)", point.x, point.y);
 
-    RetainPtr window = [m_view window];
+    RetainPtr window = [m_view.get() window];
```
We should store `m_view.get()` in a local variable

cdumez commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -4944,7 +4966,7 @@ static NSPasteboardName pasteboardNameForAccessCategory(WebCore::DOMPasteAccessC
     suppressContentRelativeChildViews(ContentRelativeChildViewsSuppressionType::TemporarilyRemove);
 
-    WebCore::FloatPoint viewCenter(NSMidX([m_view bounds]), NSMidY([m_view bounds]));
+    WebCore::FloatPoint viewCenter(NSMidX([m_view.get() bounds]), NSMidY([m_view.get() bounds]));
```
We should store `m_view.get()` in a local to avoid double ref'ing.

cdumez commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -4808,9 +4830,9 @@ static NSPasteboardName pasteboardNameForAccessCategory(WebCore::DOMPasteAccessC
         windowCaptureRect = boundsForCustomSwipeViews;
     else {
-        FloatRect unobscuredBounds = [m_view bounds];
+        FloatRect unobscuredBounds = [m_view.get() bounds];
```
We should store `m_view.get()` in a local variable

cdumez reviewed (approved):
LGTM with fixes.

csaavedra commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -6907,19 +6933,19 @@ static NSTextAlignment nsTextAlignmentFromTextAlignment(TextAlignment textAlignm
         m_imageAnalyzerQueue = WorkQueue::create("WebKit image analyzer queue"_s);
         m_imageAnalyzer = createImageAnalyzer();
-        [m_imageAnalyzer setCallbackQueue:m_imageAnalyzerQueue->dispatchQueue()];
+        [m_imageAnalyzer setCallbackQueue:RetainPtr { m_imageAnalyzerQueue->dispatchQueue() }.get()];
```
The queue can also be marked as `const`, so this shouldn't be needed.

csaavedra committed:
Address safer CPP warnings in WebViewImpl.mm
https://bugs.webkit.org/show_bug.cgi?id=299872

Reviewed by Chris Dumez.

Incidentally also fixed a few warnings in PageClientImplMac.mm, but
that file still has plenty of its own.

Renamed some methods to add a singleton suffix, others were made into
protected variants, added a few protected getters, etc.

* Source/WebKit/SaferCPPExpectations/RetainPtrCtorAdoptCheckerExpectations:
* Source/WebKit/Shared/mac/PasteboardTypes.h:
* Source/WebKit/Shared/mac/PasteboardTypes.mm:
(WebKit::PasteboardTypes::forEditingSingleton):
(WebKit::PasteboardTypes::forURLSingleton):
(WebKit::PasteboardTypes::forImagesSingleton):
(WebKit::PasteboardTypes::forImagesWithArchiveSingleton):
(WebKit::PasteboardTypes::forSelectionSingleton):
(WebKit::PasteboardTypes::forEditing): Deleted.
(WebKit::PasteboardTypes::forURL): Deleted.
(WebKit::PasteboardTypes::forImages): Deleted.
(WebKit::PasteboardTypes::forImagesWithArchive): Deleted.
(WebKit::PasteboardTypes::forSelection): Deleted.
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::isFullScreen):
(WebKit::PageClientImpl::enterFullScreen):
(WebKit::PageClientImpl::exitFullScreen):
(WebKit::PageClientImpl::beganEnterFullScreen):
(WebKit::PageClientImpl::beganExitFullScreen):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(-[WKAccessibilitySettingsObserver _settingsDidChange:]):
(-[WKWindowVisibilityObserver _windowDidOrderOnScreen:]):
(-[WKWindowVisibilityObserver _windowDidOrderOffScreen:]):
(-[WKWindowVisibilityObserver _windowDidBecomeKey:]):
(-[WKWindowVisibilityObserver _windowDidResignKey:]):
(-[WKWindowVisibilityObserver _windowDidMiniaturize:]):
(-[WKWindowVisibilityObserver _windowDidDeminiaturize:]):
(-[WKWindowVisibilityObserver _windowDidMove:]):
(-[WKWindowVisibilityObserver _windowDidResize:]):
(-[WKWindowVisibilityObserver _windowWillBeginSheet:]):
(-[WKWindowVisibilityObserver _windowDidChangeBackingProperties:]):
(-[WKWindowVisibilityObserver _windowDidChangeScreen:]):
(-[WKWindowVisibilityObserver _windowDidChangeOcclusionState:]):
(-[WKWindowVisibilityObserver _windowWillClose:]):
(-[WKWindowVisibilityObserver _screenDidChangeColorSpace:]):
(-[WKWindowVisibilityObserver observeValueForKeyPath:ofObject:change:context:]):
(-[WKWindowVisibilityObserver _activeSpaceDidChange:]):
(-[WKDOMPasteMenuDelegate _web_grantDOMPasteAccess]):
(-[WKTextTouchBarItemController textListViewController]):
(-[WKImageAnalysisOverlayViewDelegate initWithWebViewImpl:]):
(-[WKImageAnalysisOverlayViewDelegate dealloc]):
(-[WKImageAnalysisOverlayViewDelegate observeValueForKeyPath:ofObject:change:context:]):
(-[WKImageAnalysisOverlayViewDelegate firstResponderIsInsideImageOverlay]):
(-[WKImageAnalysisOverlayViewDelegate contentsRectForImageAnalysisOverlayView:]):
(WebKit::m_flagsChangedEventMonitorTrackingArea):
(WebKit::WebViewImpl::window):
(WebKit::WebViewImpl::protectedWindow):
(WebKit::WebViewImpl::handleProcessSwapOrExit):
(WebKit::WebViewImpl::setDrawsBackground):
(WebKit::WebViewImpl::setBackgroundColor):
(WebKit::WebViewImpl::becomeFirstResponder):
(WebKit::WebViewImpl::resignFirstResponder):
(WebKit::WebViewImpl::showWarningView):
(WebKit::WebViewImpl::isFocused const):
(WebKit::WebViewImpl::createPDFHUD):
(WebKit::WebViewImpl::renewGState):
(WebKit::WebViewImpl::setFrameSize):
(WebKit::WebViewImpl::setFrameAndScrollBy):
(WebKit::WebViewImpl::updateWindowAndViewFrames):
(WebKit::WebViewImpl::updateLayer):
(WebKit::WebViewImpl::updateViewExposedRect):
(WebKit::WebViewImpl::setIntrinsicContentSize):
(WebKit::WebViewImpl::intrinsicDeviceScaleFactor const):
(WebKit::WebViewImpl::windowDidBecomeKey):
(WebKit::WebViewImpl::windowDidResignKey):
(WebKit::WebViewImpl::windowDidChangeScreen):
(WebKit::WebViewImpl::mightBeginDragWhileInactive):
(WebKit::WebViewImpl::acceptsFirstMouse):
(WebKit::WebViewImpl::shouldDelayWindowOrderingForEvent):
(WebKit::WebViewImpl::windowResizeMouseLocationIsInVisibleScrollerThumb):
(WebKit::WebViewImpl::viewWillMoveToWindowImpl):
(WebKit::WebViewImpl::viewDidMoveToWindow):
(WebKit::WebViewImpl::viewDidChangeBackingProperties):
(WebKit::WebViewImpl::pageDidScroll):
(WebKit::WebViewImpl::scrollViewFrame):
(WebKit::WebViewImpl::updateTitlebarAdjacencyState):
(WebKit::WebViewImpl::hitTest):
(WebKit::WebViewImpl::colorSpace):
(WebKit::WebViewImpl::updateSecureInputState):
(WebKit::WebViewImpl::notifyInputContextAboutDiscardedComposition):
(WebKit::WebViewImpl::fullScreenWindowController):
(WebKit::WebViewImpl::protectedFullScreenWindowController):
(WebKit::WebViewImpl::executeSavedCommandBySelector):
(WebKit::WebViewImpl::registerEditCommand):
(WebKit::WebViewImpl::clearAllEditCommands):
(WebKit::WebViewImpl::validRequestorForSendAndReturnTypes):
(WebKit::WebViewImpl::selectionDidChange):
(WebKit::protectedMenuItem):
(WebKit::WebViewImpl::validateUserInterfaceItem):
(WebKit::WebViewImpl::setUserInterfaceItemState):
(WebKit::WebViewImpl::handleRequestedCandidates):
(WebKit::WebViewImpl::handleAcceptedCandidate):
(WebKit::WebViewImpl::preferencesDidChange):
(WebKit::WebViewImpl::dismissContentRelativeChildWindowsWithAnimation):
(WebKit::WebViewImpl::dismissContentRelativeChildWindowsWithAnimationFromViewOnly):
(WebKit::WebViewImpl::dismissContentRelativeChildWindowsFromViewOnly):
(WebKit::WebViewImpl::hasContentRelativeChildViews const):
(WebKit::WebViewImpl::suppressContentRelativeChildViews):
(WebKit::WebViewImpl::restoreContentRelativeChildViews):
(WebKit::WebViewImpl::quickLookWithEvent):
(WebKit::WebViewImpl::setAllowsLinkPreview):
(WebKit::WebViewImpl::immediateActionAnimationControllerForHitTestResult):
(WebKit::WebViewImpl::prepareForImmediateActionAnimation):
(WebKit::WebViewImpl::cancelImmediateActionAnimation):
(WebKit::WebViewImpl::completeImmediateActionAnimation):
(WebKit::WebViewImpl::didChangeContentSize):
(WebKit::WebViewImpl::videoControlsManagerDidChange):
(WebKit::WebViewImpl::setIgnoresNonWheelEvents):
(WebKit::WebViewImpl::accessibilityRegisterUIProcessTokens):
(WebKit::WebViewImpl::accessibilityAttributeValue):
(WebKit::WebViewImpl::updatePrimaryTrackingAreaOptions):
(WebKit::WebViewImpl::removeTrackingRect):
(WebKit::WebViewImpl::toolTipOwnerForSendingMouseEvents const):
(WebKit::WebViewImpl::sendToolTipMouseExited):
(WebKit::WebViewImpl::sendToolTipMouseEntered):
(WebKit::WebViewImpl::toolTipChanged):
(WebKit::WebViewImpl::updateThumbnailViewLayer):
(WebKit::WebViewImpl::setInspectorAttachmentView):
(WebKit::WebViewImpl::sendDragEndToPage):
(WebKit::WebViewImpl::draggingEntered):
(WebKit::WebViewImpl::draggingUpdated):
(WebKit::WebViewImpl::draggingExited):
(WebKit::handleLegacyFilesPromisePasteboard):
(WebKit::WebViewImpl::performDragOperation):
(WebKit::WebViewImpl::hitTestForDragTypes):
(WebKit::WebViewImpl::registerDraggedTypes):
(WebKit::WebViewImpl::didPerformDragOperation):
(WebKit::WebViewImpl::startWindowDrag):
(WebKit::WebViewImpl::startDrag):
(WebKit::WebViewImpl::setPromisedDataForImage):
(WebKit::WebViewImpl::provideDataForPasteboard):
(WebKit::pasteboardForAccessCategory):
(WebKit::WebViewImpl::requestDOMPasteAccess):
(WebKit::WebViewImpl::takeViewSnapshot):
(WebKit::WebViewImpl::setMagnification):
(WebKit::WebViewImpl::swipeWithEvent):
(WebKit::WebViewImpl::magnifyWithEvent):
(WebKit::WebViewImpl::smartMagnifyWithEvent):
(WebKit::WebViewImpl::gestureEventWasNotHandledByWebCore):
(WebKit::WebViewImpl::gestureEventWasNotHandledByWebCoreFromViewOnly):
(WebKit::WebViewImpl::collectKeyboardLayoutCommandsForEvent):
(WebKit::WebViewImpl::interpretKeyEvent):
(WebKit::WebViewImpl::doCommandBySelector):
(WebKit::WebViewImpl::firstRectForCharacterRange):
(WebKit::WebViewImpl::characterIndexForPoint):
(WebKit::WebViewImpl::inputContext):
(WebKit::WebViewImpl::performKeyEquivalent):
(WebKit::WebViewImpl::keyDown):
(WebKit::WebViewImpl::nativeMouseEventHandler):
(WebKit::WebViewImpl::removeFlagsChangedEventMonitor):
(WebKit::WebViewImpl::mouseMoved):
(WebKit::WebViewImpl::windowIsFrontWindowUnderMouse):
(WebKit::WebViewImpl::userInterfaceLayoutDirection):
(WebKit::WebViewImpl::effectiveAppearanceIsDark):
(WebKit::WebViewImpl::useFormSemanticContext const):
(WebKit::WebViewImpl::updateTouchBar):
(WebKit::WebViewImpl::dismissTextTouchBarPopoverItemWithIdentifier):
(WebKit::WebViewImpl::updateTextTouchBar):
(WebKit::WebViewImpl::updateMediaTouchBar):
(WebKit::WebViewImpl::shouldRequestCandidates const):
(WebKit::WebViewImpl::updateCursorAccessoryPlacement):
(WebKit::WebViewImpl::ensureImageAnalyzer):
(WebKit::WebViewImpl::installImageAnalysisOverlayView):

Canonical link: https://commits.webkit.org/301130@main
